### PR TITLE
Configs refactor

### PR DESCRIPTION
**Pull request by Razz4780**
Implemented a new class for managing mirrord configurations in the current workspace.

1. Added `selectActiveConfig` command, which allows the user to select a temporary active config override for the whole workspace. This config will be used for all mirrord invokations. All files across workspace are selectable, as long as their names end with `mirrord.{json,toml,yml,yaml}.
2. Allowed specifying mirrord config per launch configuration via the `MIRRORD_CONFIG_FILE` environment variable.
3. If no config is specified when we inject into the debugging process, the extension tries to find a default config in the `.mirrord` directory of the workspace folder (debug configurations come to us with the workspace folder specified)
4. `changeSettings` command now allows the user to select the config to edit. The user can edit the active config, configs specified in launch configurations and default configs.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -19,56 +19,250 @@ export const DEFAULT_CONFIG = `
...
+     * Default configs are located in the `.mirrord` directory and their names end with `mirrord.{toml,json,yml,yaml}`.
+     * If there are multiple candidates, they are sorted according alphabetically and the first one is returned.
+     * @param folder searched workspace folder
+     * @returns path to the found config
+     */
+    private static async getDefaultConfig(folder: vscode.WorkspaceFolder): Promise<vscode.Uri | undefined> {
+        let pattern = new vscode.RelativePattern(folder, ".mirrord/*mirrord.{toml,json,yml,yaml}");
+        let files = await vscode.workspace.findFiles(pattern);
+        files.sort((f1, f2) => {
+            if (f1.path > f1.path) {
```
Comment (eyalb181) [inline src/config.ts:96]:
```suggestion
            if (f1.path > f2.path) {
```

Comment (eyalb181) [reply src/config.ts:96]: Also, I think you can replace this whole block with `f1.path.localeCompare(f2.path)`

File: src/config.ts [hunk 2]
```diff
@@ -19,56 +19,250 @@ export const DEFAULT_CONFIG = `
...
+            let path = await MirrordConfigManager.getDefaultConfig(folder);
+            if (path) {
+                options.set(`(default) ${vscode.workspace.asRelativePath(path)}`, path);
+            } else {
+                let path = vscode.Uri.joinPath(folder.uri, ".mirrord/mirrord.json");
+                options.set(`(create default) ${vscode.workspace.asRelativePath(path)}`, path);
+            }
+        }
+
+        let quickPickOptions = [...options.keys()];
```
Comment (eyalb181) [inline src/config.ts:172]: What if there's just one option or no options? Can't we just open it immediately? Same as we do today

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 2 above)
Comment (Razz4780) [reply src/config.ts:172]: Nice catch, thanks

File: README.md [hunk 3]
```diff
@@ -27,7 +27,15 @@ mirrord main repository can be found [here](https://github.com/metalbear-co/mirr
...
+mirrord allows for rich configuration of the environment it provides. The schema for is documented [here](https://mirrord.dev/docs/overview/configuration/). You can also use `toml` or `yaml` format. However, the extension supports autocomplete only for `json` files.
+
+mirrord reads its configuration from following locations:
+
+1. Active config can be set for the whole workspace using the `selectActiveConfig` command or the link in the dropdown menu. If active config is set, mirrord always uses it.
+2. If active config is not set, mirrord searches process environment (specified in launch configuration) for `MIRRORD_CONFIG_FILE` variable. This can be an absolute path or path relative to the project root.
+3. If no config is specified, mirrord uses a default project config. This config is located in `.mirrord` directory and its name ends with `mirrord.{json,toml,yaml,yml}`. If there is no default config, mirrord creates a minimalistic one at `.mirrord/mirrord.json` when needed. If there are many candidates for default configs, mirrord sorts them alphabetically and uses the first one.
```
Comment (eyalb181) [inline README.md:39]:
```suggestion
3. If no config is specified, mirrord looks for a default project config file in the `.mirrord` directory with a name ending with `mirrord.{json,toml,yaml,yml}`. If there is no default config file, mirrord uses default configuration values for everything. If there are many candidates for the default config file, mirrord sorts them alphabetically and uses the first one.
```

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**